src/Compilers/CSharp/Portable/Binder/Binder_InterpolatedString.cs

```diff
@@ -271,26 +280,119 @@ private BoundInterpolatedString BindUnconvertedInterpolatedStringToHandlerType(B
+            if (!overloadResolutionSucceeded)
+            {
+                // MakeConstructorInvocation can call CoerceArguments on the builder if overload resolution succeeded ignoring accessibility, which
+                // could still end up not succeeding, and that would end up changing the arguments. So we want to clear and repopulate.
+                argumentsBuilder.Clear();
+
+                // Try again with an out parameter. Note that we intentionally use `diagnostics` for resolving System.Boolean, because we want to
+                // track that we're using the type no matter what.
+                var boolType = GetSpecialType(SpecialType.System_Boolean, diagnostics, unconvertedInterpolatedString.Syntax);
+                var trailingConstructorValidityPlaceholder = new BoundInterpolatedStringArgumentPlaceholder(unconvertedInterpolatedString.Syntax, BoundInterpolatedStringArgumentPlaceholder.TrailingConstructorValidityParameter, boolType);
+                var outConstructorAdditionalArguments = additionalConstructorArguments.NullToEmpty().Add(trailingConstructorValidityPlaceholder);
+                populateArguments(unconvertedInterpolatedString, outConstructorAdditionalArguments, baseStringLength, numFormatHoles, intType, argumentsBuilder);
+
+                refKindsBuilder.Add(RefKind.Out);
+
+                var outConstructorDiagnostics = BindingDiagnosticBag.GetInstance(template: diagnostics);
+                var outConstructorCall = MakeConstructorInvocation(interpolatedStringHandlerType, argumentsBuilder, refKindsBuilder, unconvertedInterpolatedString.Syntax, outConstructorDiagnostics, out overloadResolutionSucceeded);
+
+                if (overloadResolutionSucceeded)
+                {
+                    // We successfully bound the out version, so set all the final data based on that binding
+                    constructorCall = outConstructorCall;
+                    diagnostics.AddRangeAndFree(outConstructorDiagnostics);
+                    nonOutConstructorDiagnostics.Free();
+                    additionalConstructorArguments = outConstructorAdditionalArguments;
+                }
+                else
+                {
+                    // We'll attempt to figure out which failure was "best" by looking to see if one failed to bind because it couldn't find
+                    // a constructor with the correct number of arguments. We presume that, if one failed for this reason and the other failed
+                    // for a different reason, that different reason is the one the user will want to know about. If both or neither failed
+                    // because of this error, we'll report everything.
+
+                    var nonOutConstructorHasArityError = nonOutConstructorDiagnostics.DiagnosticBag?.AsEnumerableWithoutResolution().Any(d => (ErrorCode)d.Code == ErrorCode.ERR_BadCtorArgCount) ?? false;
```

Comment:
It feels weird to inspect the diagnostics to figure out what happened. Can we expose this logic from the overload resolution result and report it from `TryPerformConstructorOverloadResolution` ?

Reply:
No, that doesn't work here. For example, if there's a `ref` mismatch on one both results will get a resultkind of `OverloadResolutionFailure`, but the refkind mismatch error is the better one to report.